### ISSUE TITLE
Why does this display confusion occur?

### ISSUE THREAD

**Issue by linruicheng1997** [openseadragon/openseadragon#2161]
![image](https://user-images.githubusercontent.com/53200445/168425632-eaee98b8-1f2f-4ba4-b9ce-86655ee85022.png)

Why does the OSD display such a disordered display?

**Comment by iangilman** [comment]
Good question! It certainly doesn't usually do that. Can you share more about your project, or share a link to it?

**Comment by linruicheng1997** [comment]
Sorry, there is no link for this at the moment, I mainly want to do a function of viewing pyramid images.
I should have found the reason, this happens when my server takes more than 100ms to load a Tile. But when my server loads a Tile in less than 50ms, it is very smooth. I am now thinking about how to optimize my server-side code to make it faster to read jpg or png images. I use the OpenSlide tool to read the image!

**Comment by linruicheng1997** [comment]
Sorry, there is no link for this at the moment, I mainly want to do a function of viewing pyramid images.
I should have found the reason, this happens when my server takes more than 100ms to load a Tile. But when my server loads a Tile in less than 50ms, it is very smooth. I am now thinking about how to optimize my server-side code to make it faster to read jpg or png images. I use the OpenSlide tool to read the image!

**Comment by linruicheng1997** [comment]
> Good question! It certainly doesn't usually do that. Can you share more about your project, or share a link to it?

Sorry, there is no link for this at the moment, I mainly want to do a function of viewing pyramid images.
I should have found the reason, this happens when my server takes more than 100ms to load a Tile. But when my server loads a Tile in less than 50ms, it is very smooth. I am now thinking about how to optimize my server-side code to make it faster to read jpg or png images. I use the OpenSlide tool to read the image!

**Comment by linruicheng1997** [comment]
Do you have any good suggestions? Or do you have the source code of the server for me to learn from. Grateful

**Comment by darwinjob** [comment]
I've seen something similar before. Could you please post your browser network tab to see how the failed tiles actually fail?

**Comment by iangilman** [comment]
Hmm... It's still strange that a tile loading delay would misalign a tile! I wonder why that's happening. 

As for making your server faster, that's outside of my knowledge. Do you know what's causing the slowness? Are you tiling the images on the fly? You might consider caching them or tiling them ahead of time.

**Comment by linruicheng1997** [comment]
> 我以前见过类似的东西。您能否发布您的浏览器网络选项卡以查看失败的磁贴实际上是如何失败的？

![image](https://user-images.githubusercontent.com/53200445/169179195-557eb422-cfd4-462d-a5fd-2f629bfd3714.png)
![image](https://user-images.githubusercontent.com/53200445/169179220-94089b9d-6260-4ba4-8710-b60e0409d9b3.png)

OK, but I'm not sure if this is what you're talking about, and if so, the browser doesn't throw any errors!

**Comment by linruicheng1997** [comment]
> 嗯...瓷砖加载延迟会使瓷砖错位仍然很奇怪！我想知道为什么会这样。
> 
> 至于让你的服务器更快，那是我不知道的。你知道是什么导致缓慢吗？您是否正在动态平铺图像？您可能会考虑提前缓存它们或平铺它们。

Hello, yes I am using dynamic tiling. I haven't used the cache yet, I'm not sure what is the condition for the early cache to trigger? Do you want to cache all the tiles at one time? If not, the first time the front-end is loaded and then put into the cache, the same problem will also occur in the first display. But these are my guesses, I will quickly verify this suggestion of yours, thank you!

**Comment by darwinjob** [comment]
> OK, but I'm not sure if this is what you're talking about, and if so, the browser doesn't throw any errors!

Yeah, it seems all the tiles are actually loaded but misplaced. Is the error reproducible?

**Comment by linruicheng1997** [comment]
Sorry, I found the problem. This is not caused by OSD, but when my server generates Tile, there is already a problem with the location. By modifying my server code, I have now solved it! thank you all

**Comment by iangilman** [comment]
Fantastic! Thank you for letting us know :-)

For what it's worth, when I was mentioning the cache, I was suggesting using a server side cache, not something for the client. OpenSeadragon and the browser already automatically cache things as needed on the client.
